llvm/test/tools/llvm-foreach/llvm-foreach-lin.ll

```diff
@@ -4,15 +4,22 @@
 ; RUN: echo 'Content of second file' > %t2.tgt
 ; RUN: echo "%t1.tgt" > %t.list
 ; RUN: echo "%t2.tgt" >> %t.list
-; RUN: llvm-foreach --in-replace="{}" --in-file-list=%t.list -- echo "{}" > %t.res
+; RUN: llvm-foreach --jobs=2 --in-replace="{}" --in-file-list=%t.list -- echo "{}" > %t.res
 ; RUN: FileCheck < %t.res %s
-; CHECK: [[FIRST:.+1.tgt]]
-; CHECK: [[SECOND:.+2.tgt]]
-;
+; CHECK-DAG: [[FIRST:.+1.tgt]]
+; CHECK-DAG: [[SECOND:.+2.tgt]]
+
+; RUN: llvm-foreach --in-replace="{}" --in-file-list=%t.list -- echo "{}" > %t.res
+; RUN: FileCheck < %t.res %s --check-prefix=CHECK-ORDER
+; CHECK-ORDER: [[FIRST:.+1.tgt]]
+; CHECK-ORDER: [[SECOND:.+2.tgt]]
+
 ; RUN: llvm-foreach --in-replace="{}" --out-replace=%t --out-ext=out --in-file-list=%t.list --out-file-list=%t.out.list -- cp "{}" %t
 ; RUN: FileCheck < %t.out.list %s --check-prefix=CHECK-LIST
-; CHECK-LIST: [[FIRST:.+\.out]]
-; CHECK-LIST: [[SECOND:.+\.out]]
+; RUN: llvm-foreach --jobs=2 --in-replace="{}" --out-replace=%t --out-ext=out --in-file-list=%t.list --out-file-list=%t.out.list -- cp "{}" %t
+; RUN: FileCheck < %t.out.list %s --check-prefix=CHECK-LIST
+; CHECK-LIST-DAG: [[FIRST:.+\.out]]
```

Comment:
So, with the new parallel functionality, the results may come out in a different order, correct? I don't remember exactly how tables that we use under clang driver work, but something like "all the rows should refer to the same processed file" comes into mind. Will it be possible that we will break it somehow?

Reply:
I don't think we can break something with this change (BTW, it passed full CI testing with 4 threads enabled by default) -  commands execution order should not affect the result.
But if it's possible, could you please share where to check the info about tables in clang driver?

Reply:
There is some info https://github.com/intel/llvm/blob/sycl/sycl/doc/CompilerAndRuntimeDesign.md#sycl-support-in-the-driver .
What I'm worrying about is if the output file list will come out having different order of content - it may break the tables. If this situation is not possible, you can ignore my comment.

Reply:
llvm-foreach can't handle file tables, I believe, only lists. But even with lists, the order of input files and corresponding output files must match. From what I can see, this is preserved, as this code
   if (!OutputFileList.empty())
        OS << Path << "\n";
is executed in the order of input file iteration, even though job results which generate output file may arrive in any order. But it would be good to have a test for this with larger number of files in a list.

Reply:
> But even with lists, the order of input files and corresponding output files must match. From what I can see, this is preserved, as this code
if (!OutputFileList.empty())
OS << Path << "\n";
is executed in the order of input file iteration, even though job results which generate output file may arrive in any order.

Reading through this comment, I wonder, why do even need `-DAG` checks then?

Reply:
My bad, haven't seen that these exact checks were about output-list, not the `echo` output. Will remove `-DAG` for this case.

Reply:
>My bad, haven't seen that these exact checks were about output-list, not the echo output. Will remove -DAG for this case.

But we do need to have a test which checks `echo` output to ensure that the output is ordered if we only have one job